### PR TITLE
avoid premature form submission when using IME

### PR DESCRIPTION
**Pull request by kun432**
### Description

When using an Japanese Input Method Editor (IME), the Enter key is used for finalizing character conversion. This conflicts with pressing the Enter key to submit form, and causes premature form submission and frustration for Japanese users who are in the middle of inputting text.

This PR ensures that the Enter key will only send the message if the user is not in the middle of an IME conversion, which should provide a more intuitive experience for users typing in Japanese.

NOTES: not sure but this might also happen in Chinese and Korean language which use IME.

### Solution

- track whether the user is in the middle of an IME conversion by using the `compositionstart` and `compositionend` events
- modify `handleKeyDown` to only send the message if the Enter key is pressed and the user is not in the middle of an IME conversion.

### Reference

- https://en.wikipedia.org/wiki/Japanese_input_method
- https://developer.mozilla.org/en-US/docs/Web/API/Element/compositionstart_event
- https://developer.mozilla.org/en-US/docs/Web/API/Element/compositionend_event

### CODE REVIEW

**Comment by willydouhard** [comment]
I never used an IME myself before, thank you for the learning! I approved the CI run and will test locally before merging.
Thank you for your contribution!

**Comment by willydouhard** [comment]
Could you rebase on main @kun432 ? this will fix your CI build

**Comment by kun432** [comment]
@willydouhard done!

**Review by willydouhard — APPROVED** [review]

**Comment by willydouhard** [comment]
CI passed and approved! I let you merge. Thank you for your contribution 🥳

**Comment by kun432** [comment]
@willydouhard 

Thank you for the approval! but seems I don't have write access to your repository and can't merge this. could you please do the merge?

**Comment by kun432** [comment]
@willydouhard

Thank you for merging my PR! I'm glad tocontribute to the project!😄

**Comment by willydouhard** [comment]
Happy to have you with us, can't wait to see what you will come up with next! 😄

**Pull request merged**
